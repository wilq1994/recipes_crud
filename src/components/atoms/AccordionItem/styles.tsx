import styled, { css } from "styled-components";

export const Header = styled.header`
  color: var(--secondary-color);
  font-size: 1.333rem;
  font-weight: 700;
  padding: 1em 1.5rem;
  box-shadow: 0 0 0 0px var(--secondary-color) inset;
  transition: background 0.3s ease, box-shadow 0.3s ease;
  cursor: pointer;
  &:hover {
    box-shadow: 0 0 0 2px var(--secondary-color) inset;
  }
`;

export const Content = styled.div`
  padding: 1rem 1.5rem;
`;

export const Wrapper = styled.section<{ isOpen: boolean }>`
  background: #fff;
  display: flex;
  flex-direction: column;
  border: 1px solid var(--secondary-color);
  margin-bottom: 2px;
  ${(props) =>
    props.isOpen &&
    css`
      ${Header} {
        background: var(--secondary-color);
        color: #fff;
      }
    `}
`;
