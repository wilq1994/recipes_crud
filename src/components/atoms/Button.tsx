import styled, { css } from "styled-components";

const Button = styled.button<{ small?: boolean; secondary?: boolean }>`
  background: transparent;
  color: var(--primary-color);
  font-weight: 700;
  font-size: 1rem;
  padding: 1em 1.5em;
  margin: 0;
  border: 0;
  outline: 0;
  border-radius: 3px;
  box-shadow: 0 0 0 1px var(--primary-color) inset;
  transition: background 0.3s ease, box-shadow 0.3s ease;
  cursor: pointer;
  min-width: 120px;
  &:hover {
    background: #fff;
    box-shadow: 0 0 0 3px var(--primary-color) inset;
  }

  ${(props) =>
    props.small &&
    css`
      padding: 0.5em 1em;
      min-width: auto;
    `}

  ${(props) =>
    props.secondary &&
    css`
        color: var(--tertiary-color);
        box-shadow: 0 0 0 1px var(--tertiary-color) inset;
        &:hover {
            box-shadow: 0 0 0 3px var(--tertiary-color) inset;
        }
    `}
`;

export default Button;
