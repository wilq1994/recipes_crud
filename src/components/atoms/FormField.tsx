import styled from "styled-components";

const FormField = styled.div`
  margin-bottom: 1.4rem;
  &:last-child {
    margin-bottom: 0;
  }

  label {
    color: var(--dark-blue-color);
    text-transform: uppercase;
    font-size: 0.75rem;
    font-weight: 400;
    font-weight: 700;
    letter-spacing: 1px;
    display: block;
    margin-bottom: 0.375rem;
  }

  input,
  textarea {
    font-family: sans-serif;
    font-size: 1rem;
    padding: 0.75em 1em;
    box-sizing: border-box;
    width: 100%;
  }

  textarea {
    height: 140px;
    resize: none;
  }
`;

export default FormField;
