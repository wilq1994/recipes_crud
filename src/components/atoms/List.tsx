import React from "react";
import styled from "styled-components";

const Wrapper = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  li {
    padding: 0.5em 0;
    & + li {
      border-top: 1px solid #e0e0e0;
    }
  }
`;

type Props = {
  list: string[];
};

const List = ({ list }: Props) => {
  return (
    <Wrapper>
      {list.map((item, id) => (
        <li key={id}>{item}</li>
      ))}
    </Wrapper>
  );
};

export default List;
