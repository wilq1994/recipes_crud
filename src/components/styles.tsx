import styled, { createGlobalStyle } from "styled-components";

export const GlobalStyles = createGlobalStyle`
    :root {
        --blue-color: #f0fafe;
        --dark-blue-color: #032330;
        --primary-color: #FF5500;
        --secondary-color: #008EB0;
        --tertiary-color: #7aa6b8;
    }

    html {
        font-size: 16px;
        height: 100%;
    }

    body {
        background: var(--blue-color);
        font-family: sans-serif;
        line-height: 1.4rem;
        padding: 0;
        margin: 0;
        height: 100%;
    }

    #root {
        display: flex;
        align-items: center;
        justify-content: center;
        height: 100%;
        padding: 2rem;
        box-sizing: border-box;
    }
`;

export const Wrapper = styled.div`
  max-width: 800px;
  width: 100%;
  margin: 0 auto;
`;

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 2.1rem;
  h1 {
    color: var(--dark-blue-color);
    font-weight: 400;
    font-size: 2.369rem;
    padding: 0;
    margin: 0;
  }
`;
