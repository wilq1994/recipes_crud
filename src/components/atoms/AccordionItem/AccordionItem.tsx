import React, { useState, PropsWithChildren } from "react";

import { Wrapper, Header, Content } from "./styles";

type Props = {
  name?: string;
};

export const AccordionItem = ({ name, children }: PropsWithChildren<Props>) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Wrapper isOpen={isOpen}>
      <Header onClick={() => setIsOpen((prevValue) => !prevValue)}>
        {name}
      </Header>
      {isOpen && <Content>{children}</Content>}
    </Wrapper>
  );
};
