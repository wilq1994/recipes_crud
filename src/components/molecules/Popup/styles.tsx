import styled from "styled-components";

import Button from "components/atoms/Button";

export const Wrapper = styled.div`
  background: rgba(3, 35, 48, 0.6);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Content = styled.div`
    background: #fff;
    max-width: 800px;
    width: 100%;
    border-radius: 3px;
    box-sizing: border-box;
    padding: 1.4rem;
    box-shadow: 0 0 40px rgba(3, 35, 48, 0.8);
    h2 {
        color: var(--dark-blue-color);
        font-weight: 400;
        font-size: 1.777rem;
        padding: 0;
        margin: 0.375rem 0;
    }
`;

export const Main = styled.div`
    padding: 1.4rem 0;
`;

export const Buttons = styled.div`
  display: flex;
  justify-content: flex-end;
  ${Button} + ${Button} {
    margin-left: 0.7rem;
  }
`;