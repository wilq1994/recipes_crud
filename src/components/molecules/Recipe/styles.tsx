import styled from "styled-components";

import Button from "components/atoms/Button";

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;

  h2 {
    color: var(--dark-blue-color);
    text-transform: uppercase;
    font-size: 0.75rem;
    font-weight: 400;
    font-weight: 700;
    letter-spacing: 1px;
  }
`;

export const Buttons = styled.div`
  display: flex;
  justify-content: flex-end;
  ${Button} + ${Button} {
    margin-left: 0.7rem;
  }
`;
