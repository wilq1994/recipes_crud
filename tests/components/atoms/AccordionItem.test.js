import React from "react";
import { shallow } from "enzyme";

import {AccordionItem} from "components/atoms/AccordionItem";
import {Header} from "components/atoms/AccordionItem/styles";

describe("AccordionItem", () => {
  it("should render and match snapshot", () => {
    const component = shallow(
      <AccordionItem name='Name'><div>Child</div></AccordionItem>
    );

    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it("should show content if clicked", () => {
    const component = shallow(
      <AccordionItem name='Name'><div>Child</div></AccordionItem>
    );
    
    component.find(Header).simulate('click');
    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
