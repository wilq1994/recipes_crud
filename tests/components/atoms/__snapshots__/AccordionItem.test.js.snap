// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem should render and match snapshot 1`] = `
<styled.section
  isOpen={false}
>
  <styled.header
    onClick={[Function]}
  >
    Name
  </styled.header>
</styled.section>
`;

exports[`AccordionItem should show content if clicked 1`] = `
<styled.section
  isOpen={true}
>
  <styled.header
    onClick={[Function]}
  >
    Name
  </styled.header>
  <styled.div>
    <div>
      Child
    </div>
  </styled.div>
</styled.section>
`;
